specifyweb/frontend/js_src/lib/components/TreeView/Tree.tsx

```diff
@@ -0,0 +1,267 @@
+import { supportsBackdropBlur } from '../Molecules/Dialog';
+import { useHighContrast, useReducedTransparency } from '../Preferences/Hooks';
+import { userPreferences } from '../Preferences/userPreferences';
+import { Conformations, fetchStats, Row, Stats } from './helpers';
+import { toggleItem } from '../../utils/utils';
+import { LocalizedString } from 'typesafe-i18n';
+import { TreeRow } from './Row';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { deserializeResource } from '../DataModel/helpers';
+import { DataEntry } from '../Atoms/DataEntry';
+import { ResourceView } from '../Forms/ResourceView';
+import { getPref } from '../InitialContext/remotePrefs';
+
+const treeToPref = {
+  Geography: 'geography',
+  Taxon: 'taxon',
+  Storage: 'storage',
+  GeologicTimePeriod: 'geologicTimePeriod',
+  LithoStrat: 'lithoStrat',
+} as const;
+
+export function Tree<SCHEMA extends AnyTree>({
+  treeDefinitionItems,
+  tableName,
+  isEditingRanks,
+  focusPath: [focusPath, setFocusPath],
+  rows,
+  actionRow,
+  conformation: [conformation, setConformation],
+  getRows,
+  ranks,
+  setFocusedRow,
+  focusRef,
+  searchBoxRef,
+  baseUrl,
+  isFirst,
+}: {
+  readonly treeDefinitionItems: RA<
+    SerializedResource<FilterTablesByEndsWith<'TreeDefItem'>>
+  >;
+  readonly tableName: SCHEMA['tableName'];
+  readonly isEditingRanks: boolean;
+  readonly focusPath: GetSet<RA<number>>;
+  readonly rows: RA<Row>;
+  readonly actionRow: Row | undefined;
+  readonly conformation: GetSet<Conformations | undefined>;
+  readonly getRows: (parentId: number | 'null') => Promise<RA<Row>>;
+  readonly ranks: RA<number>;
+  readonly setFocusedRow: (row: Row) => void;
+  readonly focusRef: React.MutableRefObject<HTMLAnchorElement | null>;
+  readonly searchBoxRef: React.RefObject<HTMLInputElement | null>;
+  readonly baseUrl: string;
+  readonly isFirst: boolean;
+}): JSX.Element {
+  const highContrast = useHighContrast();
+
+  const reduceTransparency = useReducedTransparency();
+
+  const [treeAccentColor] = userPreferences.use(
+    'treeEditor',
+    treeToPref[tableName],
+    'treeAccentColor'
+  );
+
+  const id = useId('tree-view');
+
+  const [collapsedRanks, setCollapsedRanks] = useCachedState(
+    'tree',
+    `collapsedRanks${tableName}`
+  );
+
+  const [synonymColor] = userPreferences.use(
+    'treeEditor',
+    treeToPref[tableName],
+    'synonymColor'
+  );
+
+  const statsThreshold = getPref(
+    `TreeEditor.Rank.Threshold.${tableName as 'Geography'}`
+  );
+  const getStats = React.useCallback(
+    async (nodeId: number | 'null', rankId: number): Promise<Stats> =>
+      rankId >= statsThreshold
+        ? fetchStats(`${baseUrl}/${nodeId}/stats/`)
+        : Promise.resolve({}),
+    [baseUrl, statsThreshold]
+  );
+
+  const [isSecondFocused, setIsSecondFocused] = React.useState<
+    number | undefined
+  >(undefined);
+
+  const [lastTreeClicked, setLastTreeClicked] = React.useState<
```

Comment:
you are setting this, but never reading as far as I can see.
is it just unfinished code or a bug?